topics/valkey-bundle.md

```diff
@@ -0,0 +1,173 @@
+
+    # Use the Valkey Search Module
+    FT.CREATE index ON HASH PREFIX 1 doc: SCHEMA name TEXT
+    HSET doc:1 name "valkey bundle"
+    FT.SEARCH index "bundle"
+    ```
+
+    Make sure to check out the full list of commands for all the bundle components:
+
+    1. [Valkey Commands](https://valkey.io/commands/)
+    2. [Valkey JSON Commands](https://valkey.io/commands/#json)
+    3. [Valkey Bloom Commands](https://valkey.io/commands/#bloom)
+    4. [Valkey Search Commands](https://valkey.io/commands/#search)
+
+Valkey Bundle supports more advanced setup options too including:
+
+1. **Persistent storage**
+    
+    Persistent storage allows you to save data snapshots locally. The command below is an example of how you can save a snapshot every 60 seconds if at least one write occurred. 
+    
+    ```
+    bash docker run --name my-valkey-extensions -d valkey/valkey-extensions valkey-server --save 60 1 --loglevel warning
+    ```
+    
+2. **Custom Flags with Environment Variable**
+    
+    This allows you to pass additional Valkey flags at runtime using the VALKEY_EXTRA_FLAGS environment variable. This is a flexible way for customizing behavior without needing to modify the existing image or use a custom configuration file. 
+    
+    ```
+    bash docker run --env VALKEY_EXTRA_FLAGS='--save 60 1 --loglevel warning' valkey/valkey-extensions
+    ```
+    
+3. **Use a Custom Configuration File**
+    
+    If you need full control over your Valkey settings, you can create a customer configuration file that will be used inside the container. This allows you to override all default settings through the use of your own valkey.conf file.
+    
+    ```
+    docker run -v /myvalkey/conf:/usr/local/etc/valkey --name my-valkey-extensions valkey/valkey-extensions valkey-server /usr/local/etc/valkey/valkey.conf
+    ```
+
+
+# Troubleshooting
+
+Here are some common issues that users may encounter when running the Valkey Bundle, along with ways to resolve them.
+
+1. **The container won’t start**
+    
+    Possible Causes: 
+
+    - The container name is already in use.
+    
+    - The host port is already taken by another process.
+    
+    Solutions:
+
+    - Run docker ps -a to check for existing containers with the same name. If the name already exists, create the Docker container using a different name.
+    
+    - Make sure no other process is using the same port.
+    
+2. **Can’t connect using valkey-cli**
+    
+    Possible Causes: 
+
+    - The CLI container and server container aren’t on the same network.
+    
+    - The server isn’t running or hasn’t finished starting up.
+    
+    Solutions:
+    
+    - Create and use a user-defined Docker network or run the CLI from inside the container.
+    
+3. **Modules aren’t loaded**
+    
+    Possible Causes: 
+
+    - You might be using the base valkey image instead of valkey/valkey-bundle.
+    
+    - The image didn’t load the modules due to an error.
```

Comment:
You don't provide any mitigation for this, and I don't really think it's a real trouble shouting step.